ostia-operator/pkg/apicast/apicast.go

```diff
@@ -224,3 +225,146 @@ func newProbe(path string, port int32, initDelay int32, timeout int32, period in
 		PeriodSeconds:       period,
 	}
 }
+
+func processRateLimitPolicies(limits []v1alpha1.RateLimit) PolicyChain {
+	var fixedLimiters []FixedWindowRateLimiter
+	var leakyLimiters []LeakyBucketRateLimiter
+	var connLimiters []ConnectionRateLimiter
+
+	for _, limit := range limits {
+		rl, err := parseRateLimit(limit)
+		if err != nil {
+			log.Error(fmt.Sprintf("error processing rate limit rule for %s - %s", limit.Name, err.Error()))
+			continue
+		}
+
+		switch rl.(type) {
+		case FixedWindowRateLimiter:
+			fwRl := rl.(FixedWindowRateLimiter)
+			fixedLimiters = append(fixedLimiters, fwRl)
+		case LeakyBucketRateLimiter:
+			lbRl := rl.(LeakyBucketRateLimiter)
+			leakyLimiters = append(leakyLimiters, lbRl)
+		case ConnectionRateLimiter:
+			crRl := rl.(ConnectionRateLimiter)
+			connLimiters = append(connLimiters, crRl)
+		default:
+			log.Errorf("unknown rate limit type %T. ignoring", rl)
+			continue
+		}
+	}
+
+	var config PolicyChainConfiguration
+	if len(fixedLimiters) > 0 {
+		config.FixedWindowLimiters = &fixedLimiters
+	}
+	if len(leakyLimiters) > 0 {
+		config.LeakyBucketLimiters = &leakyLimiters
+	}
+	if len(connLimiters) > 0 {
+		config.ConnectionLimiters = &connLimiters
+	}
+
+	return PolicyChain{"apicast.policy.rate_limit", config}
+}
+
+func parseRateLimit(rl v1alpha1.RateLimit) (interface{}, error) {
+	key := LimiterKey{rl.Name, "plain", "service"}
+	if rl.Source != "" {
+		switch rl.Source {
+		//TODO - Add support for more sources here, jwt etc ..
+		case "ip":
+			key.Name = "{{remote_addr}}"
+		default:
+			log.Errorf("unknown source %s", rl.Source)
+		}
+		key.NameType = "liquid"
+	}
+	// limit is common to leaky bucket and fixed window algorithms
+	if rl.Limit == "" {
+		if err := verifyConnLimiterProps(rl); err != nil {
+			return nil, errors.New("error in connection limiter syntax -" + err.Error())
+		}
+		return makeConnLimiterRateLimiter(*rl.Conn, *rl.Burst, *rl.Delay, key), nil
+	}
+
+	multiplier := 1
+	parsedLimitVal := strings.Split(rl.Limit, "/")
+	parsedReqs, err := strconv.Atoi(parsedLimitVal[0])
+	if err != nil || parsedReqs < 0 {
+		return nil, errors.New("limit value must be a non-negative integer")
+	}
+
+	if len(parsedLimitVal) == 2 {
```

Comment:
Check if you can simplify the time parsing with `time.ParseDuration`, created some example: https://play.golang.org/p/Q7-DGl8GKpV